jetty-websocket/websocket-server/src/main/java/org/eclipse/jetty/websocket/server/WebSocketUpgradeFilter.java

```diff
@@ -146,6 +147,7 @@ public WebSocketUpgradeFilter(WebSocketServerFactory factory)
 
     public WebSocketUpgradeFilter(NativeWebSocketConfiguration configuration)
     {
+        durableConfiguration = configuration != null;
```

Comment:
There is something strange about this.

There is currently no code which creates a `WebSocketUpgradeFilter` without a configuration except when the filter is added with the class.
```java
context.addFilter(WebSocketUpgradeFilter.class, "/*", EnumSet.of(DispatcherType.REQUEST));
```

Even when we have a non-durable filter we use a durable configuration, which turns out it doesn't matter as the filter is not reused.

So now we can have:
1. durable filter with durable configuration
2. durable filter with non-durable configuration
3. non-durable filter with durable configuration
4. non-durable filter with non-durable configuration

Do we really need all these cases, can't we work out if the filter is durable and make the configuration the same. Maybe we could do something like:
```java
@Override
public void init(FilterConfig config) throws ServletException
{
    ServletContextHandler servletContextHandler = ServletContextHandler.getServletContextHandler(config.getServletContext());
    ServletHandler servletHandler = servletContextHandler.getServletHandler();
    durableConfiguration = servletHandler.isDurable(this);
    ...
```